### PR TITLE
[Foundation] handle ambiguity in swift 4 mode for Data.init with byte sequences

### PR DESCRIPTION
**Pull request by phausler**
Inadvertently we broke projects using the functional passing for map and Data.init in swift 4 mode. Imperative initialization however was not affected. To resolve this we need to annotate that the specialization of the bytes sequence initializer needs to be introduced in swift 4.2 which should leave 4.0 mode still source compatible.

### CODE REVIEW

**Comment by phausler** [comment]
@swift-ci please smoke test

**Comment by phausler** [comment]
@swift-ci please smoke benchmark

**Comment by slavapestov** [comment]
Can you add a test?

**Comment by phausler** [comment]
the only part that isn't tested is the swift4 mode map(Data.init) case. is there a way I can get a test compiled as swift4 mode instead of 4.2 mode?

**Comment by phausler** [comment]
@swift-ci please smoke test

**Comment by xedin** [comment]
AFAIK you can pass both `4` and `4.2` to -swift-version

**Comment by phausler** [comment]
@swift-ci please smoke test

File: test/stdlib/TestData_Swift4.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+// REQUIRES: executable_test
+// REQUIRES: objc_interop
+
+import Foundation
+import StdlibUnittest
+
+var DataTests = TestSuite("TestDataSwift4")
+
+DataTests.test("functional map init usage") {
+	let res1 = [[UInt8(0), UInt8(1), UInt8(2)]].map(Data.init) // previously this could be done without being ambiguous (however in swift 4.2 an overload was added that makes it ambiguous as a function ref)
```
Comment (slavapestov) [inline test/stdlib/TestData_Swift4.swift]: This is what I had in mind, thanks!

**Comment by phausler** [comment]
@swift-ci please smoke test

**Comment by phausler** [comment]
@swift-ci please smoke test

**Review by parkera — APPROVED** [review]

**Comment by phausler** [comment]
@swift-ci please smoke test and merge

**Comment by parkera** [comment]
@swift-ci smoke test and merge

**Pull request merged**